### ISSUE TITLE
Create Schedule button causes error for a track request.

### ISSUE THREAD

**Issue by rishabhptr** [openSUSE/osem#1889]
1. I created a cfp(track) for a conference( note: venue/rooms not set).
2. User submitted a track request.

Now the admin tracks#index shows this : 
![screenshot from 2017-12-13 23-53-00](https://user-images.githubusercontent.com/26951824/33955386-632a2efa-e061-11e7-856e-4fecebad725a.png)

Now clicking the create schedule button shows error:
![screenshot from 2017-12-13 23-54-05](https://user-images.githubusercontent.com/26951824/33955481-b433331e-e061-11e7-8756-45648dfb417c.png)

Also on a side note, similar to #1856 the description is rendered markdown on tracks#index on both user and admin side,but that can be another issue.

**Comment by differentreality** [comment]
You are logged in with a user that is an admin, right? 
That's why you can see and click this button - admin user has superpowers :) 

Normally the suggested track would be accepted (which cannot be done without a room) and then confirmed, and then the track organizer would have access to the admin side to click the 'create schedule' button, and then things should go smoothly from there.

The check for showing, or not, that button is
`can? :create, @program.schedules.build(track: track)` ideally this should be enough, and we would need to do custom checks here of whether or not a room exists. 

@AEtherC0r3 any input on this?
I am open to elegant ideas to make this disappear.

**Comment by rishabhptr** [comment]
@differentreality  Yes, this is from admin side also you are absolutely right, it works perfectly when a room is assigned to the track after checking ability of the user to do so.

So, ideally the track request would have to be accepted(which in turn would prompt admin to assign room in the first place) and then admin would go ahead to create schedule for the room so in that case we should just check track.room to be true to display the button?

**Comment by differentreality** [comment]
@rishabhptr my point was not just that you are on the admin side, my point is that your user is an admin. The logged in user has the attribute is_admin. Admin users are allowed to access almost anything, everywhere. Which is why when checking if the user has the ability to do schedule the track, the answer is yes.

I do not want to add a big if statement there. I very  much prefer authorization to handle this. 
(But it is precisely the authorization part that allows admin users to do more stuff. Maybe it makes sense to limit the abilities of admin users when it comes to suggested tracks scheduling. @AEtherC0r3 ?)
